### PR TITLE
Add specs

### PR DESCRIPTION
**Pull request by survived**
Adds the spec to the repo. Spec is being built on CI and then published on Github Pages (https://dfns.github.io/cggmp21/cggmp21-spec.pdf) on each merge to `m`. Also, cggmp21 docs are built and published on Github Pages as well so cggmp21 spec is not left alone.

### CODE REVIEW

**Comment by maurges** [comment]
Did you delete the original repo? Now I can't check what changed between them

**Comment by survived** [comment]
The only change in the spec was made in this commit 84efa7a012fdaba461488a9e1d887ade2d2f01ac

File: spec/main.tex [hunk 1]
```diff
@@ -0,0 +1,1398 @@
...
+		$\enc_N(M; r)$ 
+		encrypts a message $M \in \{-(N-1)/2, \ldots,  (N-1)/2\}$ 
+		using randomness $r \in \Z_N^*$ 
+		and Paillier public key~$N$. 
+		This produces a ciphertext $C \in \Z^*_{N^2}$.
+		Encryption checks that $\gcd(r, N)=1$ (and raises an error if not), and then computes
+    
+		$$\enc_N(M; r) := (1 + M \cdot N) \cdot r^N \bmod N^2$$ 
+
+        We also provide a function $\enc_N(M)$ that samples uniform $r \in \Z_N^*$ and returns~$\enc_N(M; r)$.
```
Comment (maurges) [inline spec/main.tex]: Mixing tabs and spaces

Comment (survived) [reply spec/main.tex]: PR did not introduce that, it existed before

Comment (survived) [reply spec/main.tex]: Oh actually there were no tabs before in this piece of spec. But my editor thought the tabbing would be appropriate choice for some reason. Maybe it found tabs in some other place

Comment (survived) [reply spec/main.tex]: Fixed

**Review by maurges — APPROVED** [review]

**Pull request merged**